IPython/core/magics/code.py

```diff
@@ -258,25 +258,39 @@ def pastebin(self, parameter_s=''):
 
           -d: Pass a custom description. The default will say
               "Pasted from IPython".
+          -e: Pass number of days for the link to be expired.
+              The default will be 7 days.
         """
-        opts, args = self.parse_options(parameter_s, 'd:')
+        opts, args = self.parse_options(parameter_s, "d:e:")
 
         try:
             code = self.shell.find_user_code(args)
         except (ValueError, TypeError) as e:
             print(e.args[0])
             return
 
+        expiry_days = 7
+        if "e" in opts:
```

Comment:
You can drop this `if`, since you already do `opts.get` with a default.

Reply:
I was thinking about the ways that I could handle it and at last, I came up with a mixed solution. Thanks for the suggestion. And also, I add the what's new entry.